Sources/TuistCache/ContentHashing/SourceFilesContentHasher.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+import TuistCore
+
+public protocol SourceFilesContentHashing {
+    func hash(sources: [Target.SourceFile]) throws -> String
+}
+
+public final class SourceFilesContentHasher: SourceFilesContentHashing {
+    private let contentHasher: ContentHashing
+
+    // MARK: - Init
+
+    public init(contentHasher: ContentHashing = ContentHasher()) {
+        self.contentHasher = contentHasher
+    }
+
+
+    // MARK: - SourceFilesContentHashing
+
+    /// Returns a unique hash that identifies an arry of sourceFiles
+    /// First it hashes the content of every file, in a specific order, and append to every hash the compiler flags of the file
+    /// Then it hashes again all partial hashes to get a unique identifier that represents a group of source files together with their compiler flags
+    public func hash(sources: [Target.SourceFile]) throws -> String {
```

Comment:
In this function, having something [like this](https://github.com/tuist/tuist/pull/1119#discussion_r402809761) will be very beneficial because sources and resources will be the most expensive items to hash. Projects have many of them and we need to do IO for that.